src/mixins/numeral-formatable.js

```diff
@@ -0,0 +1,267 @@
+
+      switch (size.length) {
+        case 0: return [3]
+        case 1: return [this.absVal(size[0])]
+        default: return size
+      }
+    },
+    sign: {
+      get () {
+        return this.signSymbol
+      },
+      set (val) {
+        let i = 0
+
+        if (this.numeralPrefix) {
+          for (const char of val) {
+            if (allowedMasks['#'].test(char)) break
+            if (char === '+' || char === '-') break
+            i++
+          }
+          val = val.substr(i)
+        }
+
+        const negative = val && val.length > 0 && val[0] === '-'
+        const positiveSymbol = this.positiveSign ? '+' : ''
+
+        this.signSymbol = this.options.positiveOnly ? positiveSymbol
+          : negative ? '-' : positiveSymbol
+      }
+    },
+    numeralPrefix () {
+      return this.options.prefix ? this.options.prefix : ''
+    },
+    numeralSuffix () {
+      return this.options.suffix ? this.options.suffix : ''
+    }
+  },
+
+  methods: {
+    unmaskNumeralText (text) {
+      if (!text) return text
+
+      this.sign = text = String(text)
+      text = this.internalChange ? this.removeNonNumeral(text) : this.attemptNumeralCorrection(text)
+
+      return this.parseNumber(text)
+    },
+    maskNumeralText (text) {
+      if (!text) return this.numeralZero()
+      text = this.sign ? String(text).substr(1) : String(text)
+
+      const param = {
+        masked: [],
+        i: this.blockSeparator.length - 1,
+        j: this.blockSize.length - 1,
+        length: 0,
+        integer: !this.precision
+      }
+
+      // Group the digits with specified separators and digit counts.
+      for (const digit of text.split('').reverse()) {
+        if (!this.maskNumeral(digit, param)) break
+      }
+
+      return this.maskedNumeral(param.masked.reverse())
+    },
+    maskNumeral (digit, param) {
+      if (digit !== '.') {
+        if (param.integer) {
+          this.blockSize[param.j] && this.blockSeparator[param.i] && param.masked.push(digit)
+        } else param.masked.push(digit)
+      }
+
+      if (param.integer) {
+        if (!this.blockSize[param.j]) return false
+        else if (!this.blockSeparator[param.i]) param.masked.push(digit)
+        else if (!(++param.length % this.blockSize[param.j])) {
+          param.masked.push(this.oneChar(this.blockSeparator[param.i], ' '))
+          param.i > 0 && param.i--
+          param.j > 0 && param.j--
+          param.length = 0
+        }
+      } else if (digit === '.') {
+        param.masked.push(this.decimal)
+        param.integer = true
+      }
+
+      return true
+    },
+    maskedNumeral (masked) {
+      if (!masked.length) masked.splice(0, 0, '0')
+      else if (this.blockSeparator.includes(masked[0])) masked.shift()
+
+      const result = this.addPrefixSuffix(masked.join(''))
+
+      return masked[0] === '.' ? '0' + result : result
+    },
+    numeralZero () {
+      const zero = 0
+      this.sign = '+'
```

Comment:
1. Side effects in this kind of methods is not the best practice, I would move `this.sign = '+'` to `maskNumeralText` (or even higher, but that would need my deeper understanding of the code)
2. `zero.toFixed` => `0..toFixed`, remove const
3. `toFixed()`  returns string, no need to to `toString()`

Reply:
1. It's an init. Maybe it should be in `mounted ()`

Reply:
2. If it's an integer only, the dot in `'0.'` should not be displayed.
3. Well, it's a design choice I made. Originally, I used the built-in `parseInt()` and `parseFloat()` everywhere (was why last time you see the hard limit of max 20 for options). Then I dropped them because it's only a computation limit, whereas the input field is free from such limitation. So an app may not want such limitation perhaps because it's not meant to be computed.